src/test/java/com/puppycrawl/tools/checkstyle/meta/XmlMetaWriter.java

```diff
@@ -0,0 +1,159 @@
+        final Node cdataDesc = doc.createCDATASection(moduleDetails.getDescription());
+        desc.appendChild(cdataDesc);
+        checkModule.appendChild(desc);
+        createPropertySection(moduleDetails, checkModule, doc);
+        if (!moduleDetails.getViolationMessageKeys().isEmpty()) {
+            final Element messageKeys = doc.createElement("message-keys");
+            for (String msg : moduleDetails.getViolationMessageKeys()) {
+                final Element messageKey = doc.createElement("message-key");
+                messageKey.setAttribute("key", msg);
+                messageKeys.appendChild(messageKey);
+            }
+            checkModule.appendChild(messageKeys);
+        }
+
+        writeToFile(doc, moduleDetails);
+    }
+
+    private static void createPropertySection(ModuleDetails moduleDetails, Element checkModule,
+                                              Document doc) {
+        if (!moduleDetails.getProperties().isEmpty()) {
+            final Element properties = doc.createElement("properties");
+            checkModule.appendChild(properties);
+            for (ModulePropertyDetails modulePropertyDetails : moduleDetails.getProperties()) {
+                final Element property = doc.createElement("property");
+                properties.appendChild(property);
+                property.setAttribute("name", modulePropertyDetails.getName());
+                property.setAttribute("type", modulePropertyDetails.getType());
+                if (modulePropertyDetails.getDefaultValue() != null) {
+                    property.setAttribute("default-value",
+                            modulePropertyDetails.getDefaultValue());
+                }
+                if (modulePropertyDetails.getValidationType() != null) {
+                    property.setAttribute("validation-type",
+                            modulePropertyDetails.getValidationType());
+                }
+                final Element propertyDesc = doc.createElement("description");
+                propertyDesc.appendChild(doc.createCDATASection(
+                        modulePropertyDetails.getDescription()));
+                property.appendChild(propertyDesc);
+            }
+        }
+    }
+
+    private static void writeToFile(Document document, ModuleDetails moduleDetails)
+            throws TransformerException {
+        final String rootOutputPath = System.getProperty("user.dir") + "/src/main/resources";
+        String fileSeperator = System.getProperty("file.separator");
+        if (System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("win")) {
+            fileSeperator = "\\" + fileSeperator;
+        }
+        final String moduleFilePath = FILEPATH_CONVERSION
+                .matcher(moduleDetails.getFullQualifiedName())
+                .replaceAll(fileSeperator);
+        final String modifiedPath;
+        if (moduleFilePath.contains("puppycrawl")) {
```

Comment:
please recheck full name of package ... `com. .....`

Reply:
Done